src/main/groovy/lang/ObjectRange.java

```diff
@@ -450,6 +451,9 @@ public Comparable next() {
                 value = peek();
                 nextFetched = true;
             }
+            if (value == null) {
+                throw new NoSuchElementException();
+            }
```

Comment:
I think these first 2 `if` statements could be replaced by the following and would make it cleaner:
```
if (!hasNext()) {
    throw new NoSuchElementException();
}
```